src/ch03/ch03-02_Sequencias.md

```diff
@@ -0,0 +1,226 @@
+#### sort, stable-sort (sequence, test [, key function])
+
+Esta função é destrutiva, então você é de bom tom copiar a sequência
+antes da ordenação:
+
+    (sort (copy-seq seq) :test #'string<)
+
+#### find, position (foo, sequence) - obter index
+
+além de `find-if`, `find-if-not`, `position-if`, `position-if-not` *(teste sequência)*.
+Veja também os parâmetros `:key`, `:test`.
+
+~~~lisp
+(find 20 '(10 20 30))
+;; 20
+(position 20 '(10 20 30))
+;; 1
+~~~
+
+#### search (sequence-a, sequence-b)
+
+Procura por uma subsequência em sequence-b que correspondente a sequence-a.
+Retorna a posição da subsequência correspondente ou NIL. Possui os parâmetros
+`from-end`, `end1/2` entre outros.
+
+#### substitute, nsubstitute[if,if-not]
+
+#### sort, stable-sort, merge
+
+#### replace (sequence-a, sequence-b)
+
+Substitui os elmentos de sequence-a pelos elementos de sequence-b.
+
+#### remove, delete (foo sequence)
+Faz uma cópia de sequence sem elementos que dêem match em foo. Possui os parâmetros
+:start/end`, `:key` e `:count`.
+
+`delete` faz o mesmo, pórem, dependendo do contexto ela pode destruir a sequência
+original no processo.
+
+~~~lisp
+(remove "foo" '("foo" "bar" "foo") :test 'equal)
+;; => ("bar")
+~~~
+
+Veja também `remove-if[-not]` a seguir.
+
+### mapping (map, mapcar, remove-if[-not],...)
+
+Se você já está familiarizado com map e filter em outras linguagens,
+provavelmete o que você procura é `mapcar`. Mas ele funciona apenas em listas,
+então para iterar em vetores e produzir tanto um vetor quanto uma lista, use `(map 'list function vector)`.
+
+mapcar também aceita múltiplas listas com `&rest more-seqs`. O mapeamento termina
+no momento em que a menor sequência chega ao fim.
+
+_Nota: `map` do cl21 é um `mapcar` para listas e vetores._
+
+`map` recebe o tipo de saída desejado como o primeiro argumento (`'list`, `'vector`, ou `'string`):
+
+~~~lisp
+(defparameter foo '(1 2 3))
+(map 'list (lambda (it) (* 10 it)) foo)
+~~~
+
+**Filter**, aqui, é chamado de `remove-if-not`.
+
+### Flatten a list (Alexandria)
+
+Com
+[Alexandria](https://common-lisp.net/project/alexandria/draft/alexandria.html),
+temor a função `flatten`.
+
+
+### Criando listas com variáveis
+
+Este é um dos usos de `backquote`:
+
+~~~lisp
+(defparameter *var* "bar")
+;; First try:
+'("foo" *var* "baz") ;; sem backquote
+;; => ("foo" *VAR* "baz") ;; errado
+~~~
+
+Segunda tentaiva, usando backquote interpolation.
+
+~~~lisp
+`("foo" ,*var* "baz")     ;; com backquote, vírgula
+;; => ("foo" "bar" "baz") ;; correto
+~~~
+
+O backquote avisa que nós vamos fazer a interpolação e a vírgula repassa 
+o valor da variável.
+
+Se a nossa variável é uma lista:
+
+~~~lisp
+(defparameter *var* '("bar" "baz"))
+;; First try:
+`("foo" ,*var*)
+;; => ("foo" ("bar" "baz")) ;; uma nested list
+`("foo" ,@*var*)            ;; backquote, comma-@ to
+;; => ("foo" "bar" "baz")
+~~~
+
+E. Weitz avisa que "objetos gerados desta forma muito provavelmete da mesma estrutura (ver Receita 2-7)".
```

Comment:
Isso aqui ficou incompreensível.

Reply:
`E. Weitz warns that “objects generated this way will very likely share structure (see Recipe 2-7)”.`

`E. Weitz avisa que "objetos gerados desta forma muito provavelmete vão partilhar da mesma estrutura (ver Receita 2-7)".`

Eu acho que isso basta. Mas esse Recipe me deu dúvida agora. Sabe se é parte do cookbook ou só um link?